packages/native/src/useLinking.tsx

```diff
@@ -95,7 +102,12 @@ const createMemoryHistory = () => {
 
       const id = window.history.state?.id ?? nanoid();
 
-      if (items.length) {
+      if (items.findIndex((item) => item.id === id) < 0) {
```

Comment:
This is also going to be `true` when `items.length` is`0`, causing the `else` clause in this condition to never get called. Should this check be after the `items.length` check?

Reply:
That's true. Furthermore the third statement is not needed, so I changed the conditions.